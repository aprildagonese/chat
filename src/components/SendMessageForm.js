import React from 'react'

export default class SendMessageForm extends React.Component {
  render() {
    return (
      <form className="send-message-form">
        <input placeholder="SendMessageForm"
               type="text" />
      </form>
    )
  }
}
