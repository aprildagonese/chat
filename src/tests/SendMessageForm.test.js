import React from "react";
import Enzyme, { shallow, mount } from 'enzyme'
import SendMessageForm from "../components/SendMessageForm";
import Adapter from "enzyme-adapter-react-16"

Enzyme.configure({adapter: new Adapter()})

describe('SendMessageForm', () => {
  let mockState
  let wrapper
  let mockEvent
  let sendMessageProp

  beforeEach(() => {
    sendMessageProp = jest.fn()
    wrapper = shallow(
      <SendMessageForm sendMessage={sendMessageProp}/>
    )
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('should have proper default state', () => {
    expect(wrapper.state()).toEqual({
      message: ''
    })
  })

  it('should update state with input', () => {
    wrapper.find("input").simulate("change", {
      target: {value: "hello"}
    })
    expect(wrapper.state()).toEqual({
      message: "hello"
    })
  })

  it('should clear state upon submit', () => {
    wrapper = mount(<SendMessageForm sendMessage={sendMessageProp}/>)

    wrapper.find("input").simulate("change", {
      target: {value: "hello"}
    })
    wrapper.find("input").simulate("submit", {})

    expect(wrapper.state()).toEqual({
      message: ''
    })
  })

});
