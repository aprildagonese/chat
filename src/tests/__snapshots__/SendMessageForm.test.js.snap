// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SendMessageForm should match the snapshot 1`] = `ShallowWrapper {}`;
